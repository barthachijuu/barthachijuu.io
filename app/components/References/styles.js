/**
 *
 * References Styles
 *
 * @version 1.0.0
 * @author Bartolo Amico
 * @since 27-Jul-2020
 */

import { createUseStyles } from 'react-jss';

const referencesStyle = createUseStyles({

  references: {
    background: '#1f1f1f url(images/refs.jpg) no-repeat center right',
    minHeight: '200px',
    width: '100%',
    overflow: 'hidden',
    textAlign: 'left',

    '& blockquote': {
      margin: '0 0px 30px 0px',
      paddingLeft: 0,
      position: 'relative',
      textShadow: '0px 1px 3px rgba(0, 0, 0, 1)',

      '& ::before, & ::after': {
        content: 'none',
      },

      '& p': {
        padding: 0,
        fontSize: '18px',
        lineHeight: '48px',
        color: '#ffffff',
        fontStyle: 'italic',
      },

      '& cite': {
        display: 'block',
        fontSize: '14px',
        fontStyle: 'normal',
        lineHeight: '18px',
        color: '#ffffff',

        '&::before': {
          content: `'\\2014 \\0020'`,
        },
      },
    },
  },

  textcontainer: {
    paddingTop: '96px',
    paddingBottom: '66px',
  },

  headercol: {
    paddingTop: '9px',

    '& h1': {
      fontSize: '18px',
      lineHeight: '24px',
      textAlign: 'center',
      marginBottom: '30px',
      textShadow: '0px 1px 3px rgba(0, 0, 0, 1)',
      color: '#ffffff',
      letterSpacing: '3px',
      textTransform: 'uppercase',
      fontWeight: 600,

      '& span': {
        display: 'none',
      },
    },
  },

  citation: {
    paddingRight: '10px',
    fontSize: '72px',
    lineHeight: '72px',
    textAlign: 'left',
    float: 'left', /* stylelint-disable-line */
    color: '#ffffff',
  },

  nukacontainer: {
    zoom: 1,
    position: 'relative',
    width: '82%',
  },

  nukaslider: {
    zoom: 1,
    position: 'relative',
    margin: 0,
    padding: 0,
  },

  /* mobile wide/smaller tablets */
  '@media only screen and (max-width: 769px)': { /* stylelint-disable-line */

    references: {

      '& blockquote': {
        paddingBottom: '24px',

        '& p': {
          fontSize: '20px',
          lineHeight: '42px',
        },
      },
    },

    textcontainer: {
      textAlign: 'center',
    },

    headercol: {

      '& h1': {
        fontSize: '16px',

        '& span': {
          display: 'block',
        },
      },
    },

    citation: {
      display: 'none',
    },
  },
});

export default referencesStyle;
