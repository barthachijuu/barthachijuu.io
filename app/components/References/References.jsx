/**
 *
 * References
 *
 * @version 1.0.0
 * @author Bartolo Amico
 * @since 27-Jul-2020
 */

import React from 'react';
import PropTypes from 'prop-types';
import Carousel from 'nuka-carousel';
import styles from './styles.js';


const References = (props) => {
  const classes = styles();
  const settings = {
    autoplay: true,
    decorators: [],
    framePadding: '10px',
    cellSpacing: 30,
    wrapAround: true,
    withoutControls: true,
    enableKeyboardControls: false,
    heightMode: 'current',
  };

  const getRefs = refs => (
    <div>
      <blockquote>
        <p>{refs.reference}</p>
        <cite>
          {refs.name}
          <br />
          {`${refs.position}, ${refs.company}`}
        </cite>
      </blockquote>
    </div>
  );

  return (
    <>
      <section id="refs" className={classes.references}>
        <div className={classes.textcontainer}>
          <div className="row">
            <div className={`two columns ${classes.headercol}`}>
              <h1>
                <i className={`fa fa-quote-left ${classes.citation}`} />
                <span>References</span>
              </h1>
            </div>
            <div className={`columns ${classes.nukacontainer}`}>
              <div className={classes.nukaslider}>
                <Carousel {...settings}>
                  {props.references.map((entry, index) => (
                    getRefs(entry, index)
                  ))}
                </Carousel>
              </div>
            </div>
          </div>
        </div>
      </section>
    </>
  );
};

References.propTypes = {
  // Add here some propTypes
  // and set a minimal description i.e.
  /** Description of prop "projects". */
  references: PropTypes.array.isRequired,
};

References.defaultProps = {
  // Add here some default propTypes values
};

export default References;
