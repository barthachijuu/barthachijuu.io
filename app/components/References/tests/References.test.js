/**
 *
 * Jest Unit Test for
 * References Component
 *
 */
import React from 'react';
import { shallow } from 'enzyme';
import 'mutationobserver-shim';
import References from '../References';

let component;
let baseProps;
const references = [{
  name: 'Alberto Pahle',
  position: 'Senior Interface Developer',
  company: 'Fineco Bank',
  reference: 'Lorem Ipsum.',
},
{
  name: 'Vincenzo Bonura',
  position: 'Responsible',
  company: 'Sicilian Tourist Service',
  reference: 'Lorem Ipsum Dolor Sit',
},
{
  name: 'Federico Tremarco',
  position: 'Revenue Management Planning & Forecasting ',
  company: 'Alitalia s.p.a.',
  reference: 'Lorem Ipsum Dolor Sit Cillum',
},
];
beforeAll(() => {
  baseProps = {
    router: {
    },
  };
});

beforeEach(() => {
  component = shallow(<References {...baseProps} references={references} />);
});

describe('<References />', () => {
  describe('Rendering', () => {
    it('should render self', () => {
      expect(component).toHaveLength(1);
    });
    it('should match the snapshot', () => {
      expect(component.html()).toMatchSnapshot();
    });
  });
});
