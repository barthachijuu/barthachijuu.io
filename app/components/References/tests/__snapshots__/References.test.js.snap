// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<References /> Rendering should match the snapshot 1`] = `
"<section id=\\"refs\\" class=\\"references-0-1-1\\"><div class=\\"textcontainer-0-1-2\\"><div class=\\"row\\"><div class=\\"two columns headercol-0-1-3\\"><h1><i class=\\"fa fa-quote-left citation-0-1-4\\"></i><span>References</span></h1></div><div class=\\"columns nukacontainer-0-1-5\\"><div class=\\"nukaslider-0-1-6\\"><div class=\\"slider\\" tabindex=\\"0\\" style=\\"box-sizing:border-box;display:block;height:inherit;-moz-box-sizing:border-box;position:relative;width:100%\\"><div class=\\"slider-frame\\" style=\\"box-sizing:border-box;display:block;height:100%;margin:10px;-moz-box-sizing:border-box;-ms-transform:translate(0, 0);overflow:hidden;padding:0;position:relative;touch-action:pinch-zoom pan-y;transform:translate3d(0, 0, 0);-webkit-transform:translate3d(0, 0, 0)\\"><ul class=\\"slider-list\\" style=\\"box-sizing:border-box;cursor:pointer;display:block;height:0;margin:0px -15px;padding:0;position:relative;-moz-box-sizing:border-box;-ms-transform:translate(0px, 0px);touch-action:pinch-zoom pan-y;transform:translate3d(0px, 0px, 0);-webkit-transform:translate3d(0px, 0px, 0);width:auto\\"><li class=\\"slider-slide slide-visible slide-current\\" style=\\"box-sizing:border-box;display:inline-block;height:auto;left:0;list-style-type:none;margin-bottom:auto;margin-left:15px;margin-right:15px;margin-top:auto;-moz-box-sizing:border-box;position:absolute;top:0;transform:scale(1);transition:transform .4s linear;vertical-align:top;width:0\\" tabindex=\\"-1\\"><div id=\\"Alberto Pahle\\" aria-hidden=\\"false\\" tabindex=\\"0\\"><blockquote><p>Lorem Ipsum.</p><cite>Alberto Pahle<br/>Senior Interface Developer, Fineco Bank</cite></blockquote></div></li><li class=\\"slider-slide\\" style=\\"box-sizing:border-box;display:inline-block;height:auto;left:120px;list-style-type:none;margin-bottom:auto;margin-left:15px;margin-right:15px;margin-top:auto;-moz-box-sizing:border-box;position:absolute;top:0;transform:scale(1);transition:transform .4s linear;vertical-align:top;width:0\\" tabindex=\\"-1\\"><div id=\\"Vincenzo Bonura\\" aria-hidden=\\"true\\"><blockquote><p>Lorem Ipsum Dolor Sit</p><cite>Vincenzo Bonura<br/>Responsible, Sicilian Tourist Service</cite></blockquote></div></li><li class=\\"slider-slide\\" style=\\"box-sizing:border-box;display:inline-block;height:auto;left:60px;list-style-type:none;margin-bottom:auto;margin-left:15px;margin-right:15px;margin-top:auto;-moz-box-sizing:border-box;position:absolute;top:0;transform:scale(1);transition:transform .4s linear;vertical-align:top;width:0\\" tabindex=\\"-1\\"><div id=\\"Federico Tremarco\\" aria-hidden=\\"true\\"><blockquote><p>Lorem Ipsum Dolor Sit Cillum</p><cite>Federico Tremarco<br/>Revenue Management Planning &amp; Forecasting , Alitalia s.p.a.</cite></blockquote></div></li></ul></div><style type=\\"text/css\\">.slider-slide > img { width: 100%; display: block; }
          .slider-slide > img:focus { margin: auto; }</style></div></div></div></div></div></section>"
`;
